lib/documents.js

```diff
@@ -2263,12 +2268,184 @@ function writeAllDocumentsImpl(options) {
   let operation = new Operation(
       'read forestInfo', this.client, requestOptions, 'empty', 'single'
   );
-  operation.stream = inputStream;
-  let result = requester.startRequest(operation);
-  operation.response = result;
+
+  let jobState = {
+    docInstance : this,
+    stream: inputStream,
+    requesterCount: 0,
+    docsWrittenSuccessfully:0,
+    docsFailedToBeWritten: 0,
+    jobOptions: (jobOptions)? mlutil.copyProperties(jobOptions):{}
+  };
+
+ if(jobState.jobOptions){
```

Comment:
Line 2278 ensures jobState.jobOptions will always be an object.

Can the test on jobState.jobOptions that guards this branch be removed such that the branch becomes part of the main flow.